src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -247,9 +251,20 @@ public abstract partial class JsonNamingPolicy
         public static System.Text.Json.JsonNamingPolicy CamelCase { get { throw null; } }
         public abstract string ConvertName(string name);
     }
-    public partial class JsonNode
-    {
-        public JsonNode() { }
+    public abstract partial class JsonNode
+    {
+        internal JsonNode() { }
+        public System.Text.Json.JsonElement AsJsonElement() { throw null; }
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonDocument jsonDocument) { throw null; }
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonElement jsonElement) { throw null; }
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonNode jsonNode) { throw null; }
```

Comment:
Should this be called Clone instead?

Reply:
There's `Clone` in `ICloneable` and documentation says it can "perform either a deep copy or a shallow copy", so I think keeping our method called `DeepCopy` would be more informative.

Reply:
Another thought here could be extension methods on JsonDocument and JsonElement called `ToJsonNode`.  We have some conventions in .NET that imply a copy, `To` on the source type is one, `From` on the target type is another.  Not sure we have a good convention for copy from an abstract base class: I suspect the naming here will be something that gets discussed during API review.